### PR TITLE
Agotado

### PR DESCRIPTION
**Pull request by bjornnorgaard**

### CODE REVIEW

File: Postkassedims/PostDims/PostDims.c [hunk 1]
```diff
@@ -12,37 +12,85 @@
 #include "uart.h"
 #include "lcd162.h"
 #include "smsHandler.h"
+#include "memoryHandler.h"
+#include "avr/eeprom.h"
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:18]: remove redundant newline

File: Postkassedims/PostDims/PostDims.c [hunk 2]
```diff
@@ -12,37 +12,85 @@
...
 	
+	SetupMemory();
+	
 	DDRA = 0x00;
+
+	uint8_t SRAMstring[8] = "22954785";
+	uint8_t SRAMstring2[8] = "88888888";
+	uint8_t StoredString[8];
+	LCDDispChar('A');
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:37]: redundant newlines

File: Postkassedims/PostDims/PostDims.c [hunk 3]
```diff
@@ -12,37 +12,85 @@
...
+	
+
+
+	//eeprom_read_block((void*)&StoredString, (const void*)&NonVolatileString, 8);
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:41]: remove commented code

File: Postkassedims/PostDims/PostDims.c [hunk 4]
```diff
@@ -12,37 +12,85 @@
...
+	
+	
+	LCDDispChar('D');
 	
 	
-	//SendString("ATE0\r"); // fjerner echo
-	//SendString("AT+CGMI\r"); //AT+CGMI - Hvem er manufacturer?
+	InitSMS('0', '1', "3257");
+
+	//ReplySMS("22954785");
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:54]: remove commented code

File: Postkassedims/PostDims/PostDims.c [hunk 5]
```diff
@@ -12,37 +12,85 @@
...
+	
+
+
+	//eeprom_read_block((void*)&StoredString, (const void*)&NonVolatileString, 8);
+	//LCDDispString(StoredString);
+	
+	//ReadMemory(StoredString, &NonVolatileString, 8);
+
+	//LCDDispString(StoredString);
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:47]: redundant newline

File: Postkassedims/PostDims/PostDims.c [hunk 6]
```diff
@@ -12,37 +12,85 @@
...
 	unsigned char receivedData = 0;
 	
 	int newlines = 0;
 	int flag = 0;
 	int bla = 0;
+	int bla2 = 0;
+	int bla3 = 0;
 	
 	while (1) {
 		if ((PINA & 0b00000001) == 0) {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:65]: brackets should be on new OR same line. not either or both.

File: Postkassedims/PostDims/PostDims.c [hunk 7]
```diff
@@ -12,37 +12,85 @@
...
 			if (!bla) {
-				DeleteSMS(1);
+				SaveNumber("22954785");
+				//WriteMemory("22954785", NonVolatileString, 8);
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:68]: remove comment

File: Postkassedims/PostDims/PostDims.c [hunk 8]
```diff
@@ -12,37 +12,85 @@
...
 				bla = 1;
 			}
 		} else {
 			bla = 0;
 		}
+		
+		if ((PINA & 0b00000010) == 0) {
+			if (!bla2) {
+			SaveNumber("88888888");
+			//WriteMemory(SRAMstring2, NonVolatileString, 8);
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:79]: comment

File: Postkassedims/PostDims/PostDims.c [hunk 9]
```diff
@@ -12,37 +12,85 @@
...
+			bla2 = 1;
+			}
+		}else {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:82]: missing space

File: Postkassedims/PostDims/PostDims.c [hunk 10]
```diff
@@ -12,37 +12,85 @@
...
+			bla2 = 0;
+		}
+		if ((PINA & 0b00000100) == 0) {
+			if (!bla3) {
+			ReadNumber(2);
+			//WriteMemory(SRAMstring2, NonVolatileString, 8);
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:88]: comment

File: Postkassedims/PostDims/PostDims.c [hunk 11]
```diff
@@ -12,37 +12,85 @@
...
+			bla3 = 1;
+			} else {
+			bla3 = 0;
+		}
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:92]: bad indention

File: Postkassedims/PostDims/PostDims.c [hunk 12]
```diff
@@ -12,37 +12,85 @@
...
+			ReadNumber(2);
+			//WriteMemory(SRAMstring2, NonVolatileString, 8);
+			bla3 = 1;
+			} else {
+			bla3 = 0;
+		}	
+		}
 		if (CharReady()) {
 			receivedData = ReadChar();
 			//if (receivedData != '\r' && receivedData != '\n' && newlines >4) {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/PostDims.c:96]: comment

File: Postkassedims/PostDims/memoryHandler.c [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+	
+	read_eeprom_array(address, receivedNum ,8);
+	
+	LCDDispString(receivedNum);
+	LCDDispChar('E');
+}
+
+
+
+//void WriteMemory(uint8_t *str, size_t length) {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/memoryHandler.c:52]: remove redundant code

File: Postkassedims/PostDims/memoryHandler.h [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+#ifndef MEMORYHANDLER_H_
+#define MEMORYHANDLER_H_
+
+//  Public:
+void WriteMemory(uint8_t *str, void* address, size_t length);
+void ReadMemory(uint8_t *storedString, void* address, size_t length);
+void SetupMemory();
+void SaveNumber(uint8_t *number);
+//------------------------------------------------
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/memoryHandler.h:20]: redundant newlines

File: Postkassedims/PostDims/memoryHandler.h [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+#endif /* SMSHANDLER_H_ */
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/memoryHandler.h:24]: missing endline

File: Postkassedims/PostDims/smsHandler.c [hunk 16]
```diff
@@ -49,31 +47,15 @@ void ReadSMS()
...
-	
-	
 	char number[8] = "";
-	number[0] = ReadChar();
 
-	for(int i = 1; i <8; i++)
+	for(int i = 0; i <8; i++)
 	{
 		number[i] = ReadChar();
 		//LCDDispChar(number[i]); //write received to lcd
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:55]: comment

File: Postkassedims/PostDims/smsHandler.c [hunk 17]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
 	//+CMGR: "REC UNREAD","+4522954785",,"17/03/01,14:38:36+04"
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:64]: comment

File: Postkassedims/PostDims/smsHandler.c [hunk 18]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
 	char received = 0;
 	
-	LCDDispString(number);
-
+	//LCDDispString(number);
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:67]: comment

File: Postkassedims/PostDims/smsHandler.c [hunk 19]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+	
 	SendString("AT+CMGS=");
 	SendString(number);
-	SendChar(0x0D);
-	while ((received = ReadCharWithTimeout(10000)) != 0) {
+	SendChar('\r');
+	
+	while ((received = ReadCharWithTimeout(5000)) != '>') {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:73]: make a decision about where to put the brackets and stick with it

File: Postkassedims/PostDims/smsHandler.c [hunk 20]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
 	}
-	LCDDispChar('P');
+	
+	LCDDispChar(received);
+	
 	SendString("Hello!");
 	SendChar(0x1A);
+	LCDDispChar('A');
+	
+	// HAX
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:85]: useless comment

File: Postkassedims/PostDims/smsHandler.c [hunk 21]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+	
+	char * errorMsg = "ERROR";
+	int pointer = 0;
+	char previous;
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:89]: char not initialized

File: Postkassedims/PostDims/smsHandler.c [hunk 22]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+	
+	while ((received = ReadChar()) != 0)
+	{
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:92]: bracket on new line

File: Postkassedims/PostDims/smsHandler.c [hunk 23]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+		if (received != '\r' && received != '\n') {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:93]: bracket on same line...

File: Postkassedims/PostDims/smsHandler.c [hunk 24]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+		if (previous == 'O' && received == 'K')
+		{
+			ReadChar();
+			ReadChar();
+			break;
+		} else {
+			previous = received;
+		}
+		
+		if (received == errorMsg[pointer]) {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:105]: brackets are all messed up in this file

File: Postkassedims/PostDims/smsHandler.c [hunk 25]
```diff
@@ -82,19 +64,63 @@ void ReplySMS(char* number) {
...
+		}
+	}
+	
+	_delay_ms(150);
+	
+	// END HAX
+	
+	
+	LCDDispChar('B');
 }
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:125]: redundant newline

File: Postkassedims/PostDims/smsHandler.c [hunk 26]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
 	SendString("AT+CMGD=");
 	SendInteger(index);
 	SendString("\r");
-	LCDDispChar('D');
+	
+	WaitforResponse();
+}
+
+void WaitforResponse() {
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:136]: should be WaitForResponse()

File: Postkassedims/PostDims/smsHandler.c [hunk 27]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
...
+	
+	char * errorMsg = "ERROR";
+	int pointer = 0;
+	char received;
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:140]: missing initialization

File: Postkassedims/PostDims/smsHandler.c [hunk 28]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
...
+	char previous;
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:141]: init var

File: Postkassedims/PostDims/smsHandler.c [hunk 29]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
...
+
+void SetEcho(char echo) {
+	
+	SendString("ATE");
+	SendChar(echo);
+	SendChar('\r');
+	
+	WaitforResponse();
+}
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:186]: remove newline

File: Postkassedims/PostDims/smsHandler.c [hunk 30]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
...
+		DeleteSMS(i);
+	}
+}
+
+void ConfigSMS() {
+	
+	SendString("AT+CNMI=3,1\r");
+	
+	WaitforResponse();
+
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:208]: redundant newline

File: Postkassedims/PostDims/smsHandler.c [hunk 31]
```diff
@@ -103,5 +129,88 @@ void DeleteSMS(int index) {
...
+	WaitforResponse();
+	
+}
+
+void InitSMS(char echo, char mode, char* pin) {
+	SetEcho(echo);
+	SetTextMode(mode);
+	WritePin(pin);
+	ConfigSMS();
 }
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.c:216]: missing endline

File: Postkassedims/PostDims/smsHandler.h [hunk 32]
```diff
@@ -14,6 +14,13 @@ void SendSMS(char *str);
...
 void ReplySMS(char* number);
 void DeleteSMS(int index);
+void DeleteAll(int spaces);
+void WaitforResponse();
+void WritePin(char* pin);
+void SetEcho(char echo);
+void SetTextMode(char mode);
+void InitSMS(char echo, char mode, char* pin);
+
 //------------------------------------------------
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/smsHandler.h:26]: redundant newline

File: Postkassedims/PostDims/uart.c [hunk 33]
```diff
@@ -82,8 +82,9 @@ char ReadChar()
 }
 
 char ReadCharWithTimeout(int timeout) {
+	LCDDispChar('V');
 	int counter = 0;
-	while ( (UCSRA & (1<<7)) == 0)
+	while ((UCSRA & (1<<RXC)) == 0)
```
Comment (bjornnorgaard) [inline Postkassedims/PostDims/uart.c:87]: BRACKETS!

**Review by bjornnorgaard — COMMENTED** [review]
code ugly.

File: Postkassedims/PostDims/PostDims.c (hunk 11 above)
Comment (dentych) [reply Postkassedims/PostDims/PostDims.c:92]: staver som en spasser

**Pull request closed without merging**